images.go

```diff
@@ -492,6 +504,7 @@ func (r *imageStore) SetNames(id string, names []string) error {
 		}
 		for _, name := range names {
 			if otherImage, ok := r.byname[name]; ok {
+				r.addNameToHistory(otherImage, name)
```

Comment:
I suggest calling `addNameToHistory(...)` from `removeName(...)`. This way, we're sure to not miss an execution path.

Reply:
Yes, I changed it and bound the new method to the `*Image` type, which should be more clear. :+1: